src/btree/bt_sync.c

```diff
@@ -269,8 +260,10 @@ __sync_file(WT_SESSION_IMPL *session, WT_CACHE_OP syncop)
 		__wt_txn_release_snapshot(session);
 
 	/* Clear the checkpoint flag and push the change. */
-	if (btree->checkpointing != WT_CKPT_OFF)
-		WT_PUBLISH(btree->checkpointing, WT_CKPT_OFF);
+	if (btree->checkpointing != WT_CKPT_OFF) {
+		btree->checkpointing = WT_CKPT_OFF;
+		WT_FULL_BARRIER();
+	}
```

Comment:
As mentioned in your comment above, this barrier isn't needed -- the only reason is if we want other threads to see at as soon as possible, **shrug**.

I can't find any reason we need a `WT_PUBLISH` here, though: please note that change, and let me know if I've missed something.